pkg/component/component_full_description.go

```diff
@@ -106,7 +112,12 @@ func (cfd *ComponentFullDescription) fillEmptyFields(componentDesc Component, co
 // NewComponentFullDescriptionFromClientAndLocalConfig gets the complete description of the component from both localconfig and cluster
 func NewComponentFullDescriptionFromClientAndLocalConfig(client *occlient.Client, kClient *kclient.Client, localConfigInfo *config.LocalConfigInfo, envInfo *envinfo.EnvSpecificInfo, componentName string, applicationName string, projectName string) (*ComponentFullDescription, error) {
 	cfd := &ComponentFullDescription{}
-	state := GetComponentState(client, componentName, applicationName)
+	var state State
+	if client == nil {
+		state = StateTypeNotPushed
```

Comment:
If odo can't check the actual state of the component it should not say 'Not Pushed'. We will need to introduce a new state called 'Unknown'

Reply:
done @kadel